dags/common/sftp_service.py

```diff
@@ -55,9 +57,21 @@ def __exit__(self, exc_type, exc_value, tb):
             return False
         return True
 
-    def list_files(self):
+    def list_files(self, excluded_directories=None):
         try:
-            return self.connection.listdir(self.dir)
+            file_names = []
+            self.connection.walktree(
+                self.dir,
+                lambda filename: append_file_if_not_in_excluded_directory(
+                    re.sub(self.dir + "/", "", filename),
```

Comment:
you don't need this 😄 if you check the `filename` actually it's like `upload/springer/JHEP/ftp_PUB_19-01-29_20-02-10_JHEP.zip` which contains already the `self.dir`.